### PR TITLE
feat(*): use kubernetes 1.7

### PR DESCRIPTION
**Pull request by adamreese**

### CODE REVIEW

**Comment by adamreese** [comment]
Test the image by running
```
helm init --image quay.io/adamreese/tiller:kube-v1.7
```

**Comment by thomastaylor312** [comment]
@adamreese Did you want us to review and test this now or wait until 1.7 releases?

File: glide.yaml [hunk 1]
```diff
@@ -49,12 +46,33 @@ import:
...
-  version: v0.8.0
+  version: 0.8.0
 - package: vbom.ml/util
   repo: https://github.com/fvbommel/util.git
   vcs: git
 - package: github.com/docker/distribution
-  version: ~v2.4.0
+  version: ~2.4.0
+
+# hacks for kubernetes v1.7
```
Comment (thomastaylor312) [inline glide.yaml]: Can you add something in here that clarifies why we are doing these hacks?

**Review by thomastaylor312 — COMMENTED** [review]
Code looks good with one small request. I can test this as soon as 1.7 releases

**Comment by bacongobbler** [comment]
We chatted in the dev meeting that we were testing 1.7 so we can catch things earlier in the release cycle, so this is WIP until 1.7 is released

**Comment by bacongobbler** [comment]
tested this on a v1.6.4 cluster with draft v0.5.0. No issues here!

**Comment by adamreese** [comment]
pushed up a new test image and ready for testing

**Review by technosophos — APPROVED** [review]
In addition to reviewing the code, I have tested on a variety of charts with Kube 1.5 (ACS), Kube 1.6 (Minikube) and Kube 1.7 (Minikube). So far, all works as expected. I think it is safe to merge this.

**Comment by technosophos** [comment]
NOTE: CircleCI has not been run on this last commit (according to the GitHub green box below)

**Comment by adamreese** [comment]
manually kicked off circleci

**Pull request merged**